File: app/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/data/utilities/geometry/GeneralPath.java

package com.TritiumGaming.phasmophobiaevidencepicker.data.utilities.geometry;

public final class GeneralPath extends Path2D.Float {

    public GeneralPath() {
        super(WIND_NON_ZERO, INIT_SIZE);
    }
    public GeneralPath(int rule) {
        super(rule, INIT_SIZE);
    }
    public GeneralPath(int rule, int initialCapacity) {
        super(rule, initialCapacity);
    }
    public GeneralPath(Shape s) {
        super(s, null);
    }

    GeneralPath(int windingRule,
                byte[] pointTypes,
                int numTypes,
                float[] pointCoords,
                int numCoords)
    {
        // used to construct from native

        this.windingRule = windingRule;
        this.pointTypes = pointTypes;
        this.numTypes = numTypes;
        this.floatCoords = pointCoords;
        this.numCoords = numCoords;
    }

    private static final long serialVersionUID = -8327096662768731142L;
}

File: app-java/src/main/java/com/TritiumGaming/phasmophobiaevidencepicker/utils/geometry/GeneralPath.kt
package com.TritiumGaming.phasmophobiaevidencepicker.utils.geometry

import com.TritiumGaming.phasmophobiaevidencepicker.utils.geometry.Path2D.Path2DFloat
import java.io.Serial

class GeneralPath : Path2DFloat {

    constructor() : super(WIND_NON_ZERO, INIT_SIZE)
    constructor(rule: Int) : super(rule, INIT_SIZE)
    constructor(rule: Int, initialCapacity: Int) : super(rule, initialCapacity)

    constructor(s: Shape?) : super(
        s!!, null
    )

    internal constructor(
        windingRule: Int,
        pointTypes: ByteArray?,
        numTypes: Int,
        pointCoords: FloatArray,
        numCoords: Int
    ) : super(WIND_NON_ZERO, INIT_SIZE) {
        // used to construct from native
        super.windingRule = windingRule
        this.pointTypes = pointTypes!!
        super.numTypes = numTypes
        super.floatCoords = pointCoords
        super.numCoords = numCoords
    }

    companion object {
        @Serial
        private val serialVersionUID = -8327096662768731142L
    }
}
